### PR TITLE
Feature/overpass

### PR DESCRIPTION
**Pull request by FXRoux**

### CODE REVIEW

File: src/world/map/gen/MapGenerator.cpp [hunk 1]
```diff
@@ -178,6 +181,144 @@ void MapGenerator::executeMapGenPhase(const Phase& i_Phase)
     }
 }
 
+void MapGenerator::overpass()
+{
+    int nbCaseParcourue = 1;
+    int coordX = 0;
+    int coordY = 0;
```
Comment (Feroxius) [inline src/world/map/gen/MapGenerator.cpp]: Mettre UINT

File: src/world/map/gen/MapGenerator.cpp [hunk 2]
```diff
@@ -178,6 +181,144 @@ void MapGenerator::executeMapGenPhase(const Phase& i_Phase)
     }
 }
 
+void MapGenerator::overpass()
+{
+    int nbCaseParcourue = 1;
+    int coordX = 0;
```
Comment (FXRoux) [inline src/world/map/gen/MapGenerator.cpp:187]: Change to UINT

File: src/world/map/gen/MapGenerator.cpp [hunk 3]
```diff
@@ -178,6 +181,144 @@ void MapGenerator::executeMapGenPhase(const Phase& i_Phase)
...
+    EnvType bushBiome = GRASSLAND;
+    int bushBiomeCounter = 0;
+    UINT probabilitOfBush = 7;
+
+    bool lastTileOcean = true;
+    int xAroundCoord[8] = { 0, 1,  0,  0, -1, -1, 0, 0 };
+    int yAroundCoord[8] = { 1, 0, -1, -1,  0,  0, 1, 1 };
+    int xAroundRandCoord[8] = { -1,  0,  1, -1, 1, -1, 0, 1 };
+    int yAroundRandCoord[8] = { -1, -1, -1,  0, 0,  1, 1, 1 };
+    int tempCoordX;
```
Comment (FXRoux) [inline src/world/map/gen/MapGenerator.cpp:202]: Change to UINT

File: src/world/map/gen/MapGenerator.cpp [hunk 4]
```diff
@@ -178,6 +181,144 @@ void MapGenerator::executeMapGenPhase(const Phase& i_Phase)
...
+                ++coordX;
+            }
+
+            diviseur = 2;
+        }
+
+        ++nbCaseParcourue;
+
+        // if we are outside the map nothing is done
+        if (coordX < s_Map->m_Dim.x && coordX > 0
```
Comment (FXRoux) [inline src/world/map/gen/MapGenerator.cpp:257]: Once changed to UINT no need to verify for smaller then 0

File: src/world/map/gen/MapGenerator.cpp [hunk 5]
```diff
@@ -178,6 +181,144 @@ void MapGenerator::executeMapGenPhase(const Phase& i_Phase)
...
+                // the biome to the selected tile biome
+                if (tileChange)
+                {
+                    randomNumber = randUINT(0, 7);
+
+                    tempCoordX = coordX + xAroundRandCoord[randomNumber];
+                    tempCoordY = coordY + yAroundRandCoord[randomNumber];
+
+                    // verification to make sur we're not outside the map limit
+                    if (tempCoordX < s_Map->m_Dim.x && tempCoordX > 0
```
Comment (FXRoux) [inline src/world/map/gen/MapGenerator.cpp:307]: Once changed to UINT no need to verify for smaller then 0

**Pull request closed without merging**